### PR TITLE
(WIP/WorkingCopy) Documentation updates.

### LINKED ISSUE #18: Integrate SBP, language-specific developer documentation
Add to developer wiki.
High-level explanation for non-developer audience.
Language-specific tutorials embedded in Doxygen/sphinx consumed from the serial (1).

See also: http://docs.swift-nav.com/wiki/Piksi_Integration_Tutorial

### PR DESCRIPTION
**Pull request by mookerji**
Includes:
0. Some field updates to the yaml specs and descriptions.
1. C doxygen stuff.
2. Python updates.

Closes #18, #19, #45.

/cc @mfine @fnoble

### CODE REVIEW

**Comment by mfine** [comment]
Nice, this is looking great

File: c/docs/install.dox [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+\section obtaining_source Obtaining the source
+
+The libswiftnav source and release tarballs are available from GitHub,
+https://github.com/swift-nav/libswiftnav.
+
+The latest development version of libswiftnav can be cloned from GitHub using
+the following command:
+
+    $ git clone git://github.com/swift-nav/libswiftnav.git
```
Comment (fnoble) [inline c/docs/install.dox]: s/libswiftnav/libsbp

File: c/docs/main.dox [hunk 2]
```diff
@@ -0,0 +1,10 @@
+/** \mainpage
+
+libswiftnav is a platform independent library of GNSS related functions.
```
Comment (fnoble) [inline c/docs/main.dox]: s/libswiftnav/libsbp

File: c/docs/main.dox [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+This library implements GNSS related functions and algorithms for use by
+software-defined GNSS receivers and other software.
```
Comment (fnoble) [inline c/docs/main.dox]: This description needs to be updated

File: spec/yaml/swiftnav/sbp/tracking.yaml [hunk 4]
```diff
@@ -49,7 +52,9 @@ definitions:
  - MSG_EPHEMERIS:
     id: 0x001A
     short_desc: WGS84 satellite orbit ephemeris parameters
-    desc: WGS84 satellite orbit ephemeris parameters
+    desc: |
+      A set of satellite orbit parameters that is used to calculate
+      GPS satellite positions and velocities (WGS84).
```
Comment (fnoble) [inline spec/yaml/swiftnav/sbp/tracking.yaml]: Maybe add a reference to ICD-200

Comment (fnoble) [inline spec/yaml/swiftnav/sbp/tracking.yaml]: Specifically Table 20-III

File: spec/yaml/swiftnav/sbp/piksi.yaml [hunk 5]
```diff
@@ -125,6 +126,7 @@ definitions:
 
  - MSG_THREAD_STATE:
     id: 0x0017
+    short_desc: State of the CPU threads.
```
Comment (fnoble) [inline spec/yaml/swiftnav/sbp/piksi.yaml]: s/CPU/RTOS ?

File: spec/yaml/swiftnav/sbp/piksi.yaml [hunk 6]
```diff
@@ -17,6 +17,7 @@ definitions:
 
  - MSG_PRINT:
     id: 0x0010
+    short_desc: Information and debugging information.
```
Comment (fnoble) [inline spec/yaml/swiftnav/sbp/piksi.yaml]: "Information" repeated - maybe something like "String containing errors, warnings and informational messages"?

File: spec/yaml/swiftnav/sbp/observation.yaml [hunk 7]
```diff
@@ -16,6 +16,7 @@ include:
 definitions:
 
  - ObsGPSTime:
```
Comment (fnoble) [inline spec/yaml/swiftnav/sbp/observation.yaml]: I'd vote for inlining this into ObsHeader seeing as its only used in one place

**Comment by fnoble** [comment]
sweet

**Comment by mookerji** [comment]
Supplanted by #59.

**Pull request closed without merging**